src/pages/account/mijn-historiek/index.tsx

```diff
@@ -1,13 +1,82 @@
+import { Table } from '@meemoo/react-components';
 import { GetServerSideProps, NextPage } from 'next';
 import { useTranslation } from 'next-i18next';
 import Head from 'next/head';
+import { useCallback, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { Column, TableOptions } from 'react-table';
+import { useQueryParams } from 'use-query-params';
 
+import {
+	ACCOUNT_HISTORY_QUERY_PARAM_CONFIG,
+	HistoryItemListSize,
+	HistoryTableAccessComboId,
+	HistoryTableAccessFrom,
+	HistoryTableColumns,
+} from '@account/const';
 import { AccountLayout } from '@account/layouts';
+import { selectUserProfileId } from '@auth/store/user';
+import { withAuth } from '@auth/wrappers/with-auth';
 import { withI18n } from '@i18n/wrappers';
+import { Loading, PaginationBar, sortingIcons } from '@shared/components';
+import { OrderDirection, VisitInfo, VisitStatus } from '@shared/types';
 import { createPageTitle } from '@shared/utils';
+import { useGetVisits } from '@visits/hooks/get-visits';
+import { VisitTimeframe } from '@visits/types';
 
 const AccountMyHistory: NextPage = () => {
 	const { t } = useTranslation();
+	const userProfileId = useSelector(selectUserProfileId);
+	const [filters, setFilters] = useQueryParams(ACCOUNT_HISTORY_QUERY_PARAM_CONFIG);
+
+	const visits = useGetVisits({
+		searchInput: undefined,
+		status: VisitStatus.APPROVED,
+		timeframe: VisitTimeframe.PAST,
+		page: filters.page,
+		size: HistoryItemListSize,
+		orderProp: filters.orderProp as keyof VisitInfo,
+		orderDirection: filters.orderDirection as OrderDirection,
+		userProfileId: userProfileId !== null ? userProfileId : undefined,
```

Comment:
no need to pass this param, since it defaults to the current logged in user.
```suggestion
```

Reply:
Parameter weglaten geeft resultaten voor alle gebruikers, niet de bedoeling.
![image](https://user-images.githubusercontent.com/90180417/159003954-423f69ad-4383-4423-a4c0-053a688987b8.png)